datafusion/expr/src/udf.rs

```diff
@@ -89,6 +91,34 @@ impl ScalarUDF {
             signature: signature.clone(),
             return_type: return_type.clone(),
             fun: fun.clone(),
+            aliases: vec![],
+        }
+    }
+
+    /// Create a new ScalarUDF with aliases
+    pub fn new_with_aliases(
```

Comment:
Unless there is a compelling reason, I think we should avoid adding this new constructor and only have `with_aliases` to avoid a redundant API